packages/manager/modules/pci/src/projects/project/notebooks/add/add.controller.js

```diff
@@ -1,8 +1,78 @@
-export default class {
+import get from 'lodash/get';
+import {
+  NOTEBOOK_ATTACH_STORAGE,
+  NOTEBOOK_PRIVACY_SETTINGS,
+  NOTEBOOK_RESOURCES,
+} from './notebook.constants';
+
+export default class NotebookAddController {
+  static convertLabels(labels) {
+    return labels.reduce((cum, l) => {
+      const label = { ...cum };
+      label[l.key] = l.value;
+      return label;
+    }, {});
+  }
+
+  static convertVolumes(region, storageVolumes, gitVolume) {
+    const volumes = storageVolumes.map((volume) => ({
+      swift: {
+        container: volume.container.name,
+        region: region.name,
+      },
+      mountPath: volume.mountPath,
+      permissions: volume.permission,
+      cache: true,
+    }));
+
+    if (gitVolume.repositoryURL) {
+      volumes.push({
+        git: {
+          url: gitVolume.repositoryURL,
+        },
+        mountPath: gitVolume.mountPath,
+        permissions: NOTEBOOK_ATTACH_STORAGE.PERMISSION_READ_ONLY,
+        cache: true,
+      });
+    }
+
+    return volumes;
+  }
+
+  static getCpuOrGpuResource(src, nbResources) {
+    const resource = {};
+    resource[src.flavorType] = nbResources;
+    return resource;
+  }
+
+  static createNotebook(notebook) {
+    const { name, repositoryURL, mountPath, nbResources } = notebook;
+    const { labels, volumes, selected } = notebook;
+    const { editor, framework, region, resource } = selected;
+
+    return {
+      env: {
+        editorId: editor.id,
+        frameworkId: framework.model.id,
+        frameworkVersion: framework.version.version,
+      },
+      labels: NotebookAddController.convertLabels(labels),
+      name,
+      region: region.name,
+      resources: NotebookAddController.getCpuOrGpuResource(
```

Comment:
The `getCpuOrGpuResource` method can be removed, since it is not reused?
```suggestion
      {
          [resource.flavorType]: nbResources,
      }
```

Reply:
True, fixed.